elements/pl-matrix-component-input/pl-matrix-component-input.mustache

```diff
@@ -44,15 +45,30 @@
         });
     });
 </script>
-<span id="pl-matrix-component-input-submission-{{uuid}}">
+<span id="pl-matrix-component-input-submission-{{uuid}}" class="pl-matrix-component-input-invalid">
+
+{{#parse_error}}
     {{#label}}<span>{{label}}</span>{{/label}}
-    <a role="button" class="btn btn-light btn-sm" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" title="Error" tabindex="0" data-content="{{parse_error}}{{#raw_submitted_answer}}<hr><h5><small class='text-uppercase'>What you submitted</small></h5><pre>{{raw_submitted_answer}}</pre>{{/raw_submitted_answer}}">INVALID
-      <span><i class="fa fa-question-circle" aria-hidden="true"></i></span>
-  </a>
-</span>
+
+    {{#raw_submitted_answer}}{{{raw_submitted_answer}}}{{/raw_submitted_answer}}
+    
+    <span class="badge text-danger badge-invalid"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> Invalid</span>
+    
+    <a href="javascript:void(0);" role="button" class="btn btn-sm btn-secondary small border" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" style="margin-left: 5px" title="Format Error" tabindex="0" data-content="{{parse_error}}"> Why <i class="fa fa-question-circle" aria-hidden="true"></i></a>
 {{/parse_error}}
+{{#missing_input}}
+    {{#label}}<span>{{label}}</span>{{/label}}
+
+    <span class="badge text-dark badge-missing-input"><i class="fa fa-question-circle" aria-hidden="true"></i> Missing Input</span>
+    
+    <a href="javascript:void(0);" role="button" class="btn btn-sm btn-secondary small border" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" style="margin-left: 5px" title="Missing Input" tabindex="0" data-content="There is no submitted value for this field.  This may have happened because the question was changed after the answer was submitted."> Why <i class="fa fa-question-circle" aria-hidden="true"></i>
```

Comment:
Ditto here:

How about adding "by course staff" to give context as to why the question changed?

Reply:
I like this -- makes it very clear.